harmonica/_spherical_harmonics/igrf.py

```diff
@@ -0,0 +1,696 @@
+            else:
+                variation_g = (g[index + 1, n, m] - g[index, n, m]) / epoch
+                variation_h = (h[index + 1, n, m] - h[index, n, m]) / epoch
+            g_date[n, m] = g[index, n, m] + seconds_since_epoch * variation_g
+            h_date[n, m] = h[index, n, m] + seconds_since_epoch * variation_h
+    return g_date, h_date
+
+
+def calculate_ideal_spacing(max_degree):
+    """
+    Estimate the ideal spacing for a spherical harmonic synthesis.
+
+    Uses the sampling theorem of [DriscollHealy1994]_ which provides a number
+    of points to discretize an expansion with :math:`N` maximum degree as
+    :math:`l = 2N + 2`.
+
+    Parameters
+    ----------
+    max_degree : int
+        The maximum degree of the expansion.
+
+    Returns
+    -------
+    spacing : float
+        The spacing in degrees.
+    """
+    latitude_range = 180
+    return latitude_range / (2 * max_degree + 2)
+
+
+class IGRF14:
+    r"""
+    International Geomagnetic Reference Field (14th generation).
+
+    Calculate the three components of the magnetic field vector of the IGRF
+    model in a geodetic (longitude, latitude, geometric height) system for any
+    date between 1900 and 2030. Model coefficients are automatically downloaded
+    and cached locally using :mod:`pooch`.
+
+    .. note::
+
+        The code is automatically parallelized (multithreaded shared-memory)
+        with Numba. To evaluate the model on a regular grid, it's **at least 2x
+        faster** to use the :meth:`~harmonica.IGRF14.grid` method than the
+        :meth:`~harmonica.IGRF14.predict` method since the former is able to
+        avoid repeat computations.
+
+    Parameters
+    ----------
+    date : str or :class:`datetime.datetime`
+        The date and time at which to calculate the IGRF field. If it's
+        a string, should be an `ISO 8601 formatted date
+        <https://en.wikipedia.org/wiki/ISO_8601>`__ and it will be converted
+        into a Python :class:`datetime.datetime`. Must be on or after 1900 and
+        before 2030.
+    min_degree : int
+        The minimum degree used in the expansion. Default is 1 (magnetic fields
+        don't have the 0 degree term).
+    max_degree : int
+        The maximum degree used in the expansion. Default is 13.
+    ellipsoid : :class:`boule.Ellipsoid`
+        The ellipsoid used to convert geodetic to geocentric spherical
+        coordinates and convert the magnetic field vector from a geocentric
+        spherical to a geodetic system. Default is ``boule.WGS84``.
+
+    Attributes
+    ----------
+    doi : str
+        The DOI used to download the coefficient file.
+    file_name : str
+        The name of the coefficient file in the online archive.
+    hash : str
+        The hash of the coefficient file, used to check download integrity.
+    reference_radius : float
+        The reference radius used in the spherical harmonic expansion in
+        meters.
+    coefficients : tuple = (g, h)
```

Comment:
It seems like the coefficients always have a shape of 14x14, regardless of what value is passed to `max_degree`.